### PR TITLE
Allow list of named tables to be fetched

### PR DESCRIPTION
**Pull request by melohagan**
## Description
Formula columns were being cleared on fetch tables. By being able to specify the tables you want to fetch, you can now prevent that from happening. 
In other cases you may only want to pull a subset of your tables, although this could be solved with database level schemas and multiple datasources in Budibase.

Addresses: 
- https://github.com/Budibase/budibase/discussions/5539
- https://github.com/Budibase/budibase/issues/4651

## Screenshots
1. Fetch all tables.
2. Add a formula column to the **Owners** table.
3. Within postgres add new column to the **Appointments** table
Screenshot 2022-08-17 at 18 24 30

4. Fetch tables (Appointments, Pets, Vehicles, etc)
Screenshot 2022-08-17 at 18 27 40

5. See that **Appointments** has been updated
Screenshot 2022-08-17 at 18 26 06

6. **Owners** has been preserved
Screenshot 2022-08-17 at 18 26 30

**Toggle is off by default**
Screenshot 2022-08-17 at 18 27 03

### CODE REVIEW

File: packages/server/src/api/controllers/datasource.js [hunk 1]
```diff
@@ -50,9 +50,21 @@ exports.fetch = async function (ctx) {
...
 
-  const { tables, error } = await buildSchemaHelper(datasource)
-  datasource.entities = tables
+  let { tables, error } = await buildSchemaHelper(datasource)
+  if (tablesFilter) {
+    if (!datasource.entities) {
+      datasource.entities = {}
+    }
+    for (let key in tables) {
+      if (tablesFilter.includes(key)) {
```
Comment (mike12345567) [inline packages/server/src/api/controllers/datasource.js:61]: this should probably be case insensitive - usually users don't expect to need to define a table name exactly (as some SQL databases treat identifiers as case insensitive).

**Review by mike12345567 — APPROVED** [review]
LGTM - I think the one change mentioned to make this a case insensitive search could be quite helpful.

File: packages/server/src/api/controllers/datasource.js (hunk 1 above)
Comment (melohagan) [reply packages/server/src/api/controllers/datasource.js:61]: The only risk is you could technically pull down two tables when you only want one, e.g. "appointments":

Screenshot 2022-09-13 at 10 37 34

With that said maybe it's very unlikely users would have two tables with the same name except the casing.

**Comment by melohagan** [comment]
Case insensitive table fetch added.

Screenshot 2022-09-13 at 10 52 45

Screenshot 2022-09-13 at 10 53 06

**Pull request merged**